craft_providers/lxd/launcher.py

```diff
@@ -72,48 +105,96 @@ def _create_instance(
     """
     logger.info("Creating new instance from remote")
     logger.debug(
-        "Creating new instance from image %r from remote %r.", image_name, image_remote
+        "Creating new instance from image %r from remote %r", image_name, image_remote
     )
-    instance.launch(image=image_name, image_remote=image_remote, ephemeral=ephemeral)
-    base_configuration.setup(executor=instance)
-
-    _set_timezone(instance, project, remote, lxc)
-
-    # return early if base instances and user id mapping are not specified
-    if not base_instance and not map_user_uid:
-        return
-
-    # the instance needs to be stopped before copying or updating the id map
-    instance.stop()
 
+    # Lockable base instance creation. Only one caller can create the base instance.
+    # Other callers will will get LXDError and wait until the base instance is created.
     if base_instance:
-        logger.info("Creating new base instance from instance")
+        logger.info("Preparing base instance from remote")
         logger.debug(
-            "Creating new base instance %r from instance.", base_instance.instance_name
+            "Creating new base instance from image %r from remote %r",
+            image_name,
+            image_remote,
+        )
+        base_instance.launch(
+            image=image_name,
+            image_remote=image_remote,
+            ephemeral=False,  # base instance should not ephemeral
+        )
+        base_instance_status = base_instance.config_get("user.craft_providers.status")
+
+        # Skip the base configuration if the instance is already configured.
+        if base_instance_status != "FINISHED":
```

Comment:
Can you define these statuses in one place and refer to the variable?  (enums are always a good choice for pre-defined states)

You could also make a small class and wrap `base_instance.config_get("user.craft_providers.status")` in a function.